internal/terraform/node_resource_plan.go

```diff
@@ -242,7 +260,23 @@ func (n *nodeExpandPlannableResource) expandResourceImports(ctx EvalContext, all
 
 			diags = diags.Append(validateImportTargetExpansion(n.Config, res, imp.Config.To))
 
-			importID, evalDiags := evaluateImportIdExpression(imp.Config.ID, ctx, keyData, allowUnknown)
+			var importID cty.Value
+			if imp.Config.ID != nil {
+				importID, evalDiags = evaluateImportIdExpression(imp.Config.ID, ctx, keyData, allowUnknown)
+			} else if imp.Config.Identity != nil {
+				_, providerSchema, err := getProvider(ctx, n.ResolvedProvider)
```

Comment:
if you're not using the provider, these can just be `ctx.ProviderSchema` calls.

Reply:
Oh, thanks for pointing it out again. I only replaced some of the `getProvider` calls in an earlier commit 🤦